### PR TITLE
Linux support plsplspls

### PR DESCRIPTION
**Pull request by AryToNeX**
This time it is golden! No pre-requirements, it's foolproof and, more importantly, it's safer than ever.
We also got the launcher icon right!

### CODE REVIEW

File: src/pages/Manager.vue [hunk 1]
```diff
@@ -616,23 +631,48 @@
...
+						'Steam'
+					);
+				case 'linux':
+					return path.resolve(homedir(), '.local', 'share', 'Steam');
+				default:
+					return '';
+			}
+		}
+
+		async checkIfSteamDirectoryIsValid(dir : string) : Promise<boolean> { // TODO: FIX
```
Comment (ebkr) [inline src/pages/Manager.vue:648]: Leftover TODO?

Comment (AryToNeX) [reply src/pages/Manager.vue:648]: Uhm.... yeah. it's actually done

File: src/r2mm/manager/win32/GameDirectoryResolver.ts [hunk 2]
```diff
@@ -67,7 +67,7 @@ export default class GameDirectoryResolverImpl extends GameDirectoryResolverProv
         } catch(e) {
             const err: Error = e;
             return new R2Error(
-                'Unable to resolve steam install directory',
```
Comment (ebkr) [inline src/r2mm/manager/win32/GameDirectoryResolver.ts:70]: Good spot, thanks!

File: src/r2mm/manager/linux/GameDirectoryResolver.ts [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+import * as vdf from '@node-steam/vdf';
+import * as path from 'path';
+import { homedir } from 'os';
+import ManagerSettings from '../ManagerSettings';
+import FsProvider from "../../../providers/generic/file/FsProvider";
+import GameDirectoryResolverProvider from '../../../providers/ror2/game/GameDirectoryResolverProvider';
+
+const appManifest = 'appmanifest_632360.acf';
+
+export default class GameDirectoryResolverImpl extends GameDirectoryResolverProvider {
```
Comment (ebkr) [inline src/r2mm/manager/linux/GameDirectoryResolver.ts:14]: Would be good to rename the class to something like `LinuxGameDirectoryResolver` with the file name to match

Comment (ebkr) [reply src/r2mm/manager/linux/GameDirectoryResolver.ts:14]: Same in the Windows variant too

Comment (AryToNeX) [reply src/r2mm/manager/linux/GameDirectoryResolver.ts:14]: it's namespaced, so, theoretically, it's not needed

Comment (ebkr) [reply src/r2mm/manager/linux/GameDirectoryResolver.ts:14]: It's more for IDEs referencing imports

**Review by ebkr — APPROVED** [review]

**Pull request merged**